source/parlex/src/parser.cpp

```diff
@@ -27,12 +27,12 @@ void parser::start_workers(int threadCount) {
 					//DBG("THREAD ", threadCount, " POPPING ITEM");
 					auto item = get_work_item();
 					lock.unlock();
-					context_ref const & context = *std::get<0>(item);
+					context* const & cont = std::get<0>(item);
```

Comment:
Just use `c` here instead of `cont`. I like to use the full name, or just a single letter if there is a name conflict and the single letter is easy to ascertain.